frame/referenda/src/lib.rs

```diff
@@ -1270,4 +1280,140 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 			Self::deposit_event(Event::<T, I>::MetadataCleared { index, hash });
 		}
 	}
+
+	/// Ensure the correctness of the state of this pallet.
+	///
+	/// The following assertions must always apply.
+	///
+	/// General assertions:
+	///
+	/// * `ReferendumCount` must always be equal to the number of referenda in `ReferendumInfoFor`.
+	/// * Referendum indices must be sorted in the `ReferendumInfoOf` storage map.
+	/// * Referendum indices in `MetadataOf` must also be stored in `ReferendumInfoOf`.
+	///
+	/// Looking at referenda info:
+	///
+	/// - Data regarding ongoing phase:
+	///
+	/// * The submission deposit cannot be less than `T::SubmissionDeposit`.
+	/// * There must exist track info for the track of the referendum.
+	/// * The decision deposit if provided cannot be less than the decision deposit of the track.
+	/// * The deciding stage has to begin before confirmation period.
```

Comment:
if there is enough support and approves accumulated during preparation period, referendum will move into confirmation stage skipping the decision stage.